samcli/lib/list/resources/resource_mapping_producer.py

```diff
@@ -0,0 +1,149 @@
+"""
+The producer for the 'sam list resources' command
+"""
+from typing import Any, Dict
+import dataclasses
+import logging
+import yaml
+
+from botocore.exceptions import ClientError, NoCredentialsError, BotoCoreError
+from samtranslator.translator.managed_policy_translator import ManagedPolicyLoader
+from samtranslator.translator.arn_generator import NoRegionFound
+
+from samcli.commands.list.exceptions import (
+    SamListLocalResourcesNotFoundError,
+    SamListUnknownClientError,
+    StackDoesNotExistInRegionError,
+    SamListUnknownBotoCoreError,
+)
+
+from samcli.lib.list.list_interfaces import Producer
+from samcli.lib.list.resources.resources_def import ResourcesDef
+from samcli.lib.translate.sam_template_validator import SamTemplateValidator
+from samcli.lib.providers.sam_stack_provider import SamLocalStackProvider
+from samcli.commands.validate.lib.exceptions import InvalidSamDocumentException
+from samcli.commands.local.cli_common.user_exceptions import InvalidSamTemplateException
+from samcli.commands.exceptions import UserException
+from samcli.commands._utils.template import get_template_data
+from samcli.lib.utils.boto_utils import get_client_error_code
+
+
+LOG = logging.getLogger(__name__)
+
+
+class ResourceMappingProducer(Producer):
+    def __init__(
+        self,
+        stack_name,
+        region,
+        profile,
+        template_file,
+        cloudformation_client,
+        iam_client,
+        mapper,
+        consumer,
+    ):
+        self.stack_name = stack_name
+        self.region = region
+        self.profile = profile
+        self.template_file = template_file
+        self.cloudformation_client = cloudformation_client
+        self.iam_client = iam_client
+        self.mapper = mapper
+        self.consumer = consumer
+
+    def get_resources_info(self):
+        """
+        Returns the stack resources information for the stack and raises exceptions accordingly
+
+        Returns
+        -------
+            A dictionary containing information about the stack's resources
+        """
+
+        try:
+            response = self.cloudformation_client.describe_stack_resources(StackName=self.stack_name)
+            if "StackResources" not in response:
+                return {"StackResources": []}
+            return response
+        except ClientError as e:
+            if get_client_error_code(e) == "ValidationError":
+                LOG.debug("Stack with id %s does not exist", self.stack_name)
+                raise StackDoesNotExistInRegionError(stack_name=self.stack_name, region=self.region) from e
+            LOG.error("ClientError Exception : %s", str(e))
+            raise SamListUnknownClientError(msg=str(e)) from e
+        except BotoCoreError as e:
+            LOG.error("Botocore Exception : %s", str(e))
+            raise SamListUnknownBotoCoreError(msg=str(e)) from e
+
+    def get_translated_dict(self, template_file_dict: Dict[Any, Any]) -> Dict[Any, Any]:
```

Comment:
Why we needed to translate template locally?

Reply:
We do a local translate as a way of doing a "dry run" to show what the transformed state of the stack will look like after deploy.